tests/testpathgradientbrush.c

```diff
@@ -0,0 +1,1685 @@
+    assert (status == InvalidParameter);
+
+    status = GdipGetPathGradientBlend (brush, NULL, positions, 1);
+    assert (status == InvalidParameter);
+
+    status = GdipGetPathGradientBlend (brush, blend, NULL, 1);
+    assert (status == InvalidParameter);
+
+    status = GdipGetPathGradientBlend (brush, blend, positions, 0);
+    assert (status == InvalidParameter);
+
+    status = GdipGetPathGradientBlend (brush, blend, positions, -1);
+    assert (status == InvalidParameter);
+
+    GdipDeleteBrush ((GpBrush *) brush);
+}
+
+static void test_setPathGradientBlend ()
+{
+    GpStatus status;
+    GpPathGradient *brush;
+
+    REAL blend1[1] = {3};
+    REAL positions1[1] = {-12};
+    REAL destBlend1[1];
+    REAL destPositions1[1];
+
+    REAL blend2[2] = {-1, 0};
+    REAL positions2[2] = {0, 1.0f};
+    REAL destBlend2[2];
+    REAL destPositions2[2];
+
+    REAL blend3[3] = {1, 2, 3};
+    REAL positions3[3] = {0, 0.5f, 1.0f};
+    REAL destBlend3[3];
+    REAL destPositions3[3];
+
+    REAL invalidPositions1[2] = {0.5, 1};
+    REAL invalidPositions2[2] = {0, 0.5};
+
+    ARGB pathPresetBlend[3] = {1, 2, 3};
+    REAL pathPresetPositions[3] = {0, 0.5f, 1.0f};
+    INT presetBlendCount;
+
+    GdipCreatePathGradient (threePoints, 3, WrapModeTileFlipX, &brush);
+
+    // Count of 1.
+    status = GdipSetPathGradientBlend (brush, blend1, positions1, 1);
+    assert (status == Ok);
+
+    status = GdipGetPathGradientBlend (brush, destBlend1, destPositions1, 1);
+    assert (status == Ok);
+    assert (destBlend1[0] == 3);
+    // FIXME: it appears GDI+ ignores the position value if there is a single element.
+    // assert (floatsEqual(destPositions1[0], 0));
+
+    // Count of 2.
+    status = GdipSetPathGradientBlend (brush, blend2, positions2, 2);
+    assert (status == Ok);
+
+    status = GdipGetPathGradientBlend (brush, destBlend2, destPositions2, 2);
+    assert (status == Ok);
+    assert (destBlend2[0] == -1);
+    assert (destBlend2[1] == 0);
+    assert (destPositions2[0] == 0);
+    assert (destPositions2[1] == 1);
+
+    // Count of 3.
+    status = GdipSetPathGradientBlend (brush, blend3, positions3, 3);
+    assert (status == Ok);
+
+    status = GdipGetPathGradientBlend (brush, destBlend3, destPositions3, 3);
+    assert (status == Ok);
+    assert (destBlend3[0] == 1);
+    assert (destBlend3[1] == 2);
+    assert (destBlend3[2] == 3);
+    assert (destPositions3[0] == 0);
+    assert (destPositions3[1] == 0.5);
+    assert (destPositions3[2] == 1);
+
+    // Should clear the existing blend.
+    status = GdipSetPathGradientPresetBlend (brush, pathPresetBlend, pathPresetPositions, 3);
+    assert (status == Ok);
+
+    status = GdipSetPathGradientBlend (brush, destBlend2, destPositions2, 2);
+    assert (status == Ok);
+
+    status = GdipGetPathGradientPresetBlendCount (brush, &presetBlendCount);
+    assert (status == Ok);
+    // FIXME: this returns 0 with GDI+.
+    // assert (presetBlendCount == 0);
```

Comment:
This is a libgdiplus bug that I couldn't quite figure out how to fix yet.

The problem is that a new PathGradientBrush has no presets with GDI+, but has 1 with libgdiplus.